spharpy/spherical.py

```diff
@@ -1,6 +1,431 @@
+            self._basis_inv = np.conj(self.basis).T * (weights)
+
+    @property
+    def basis_inv_gradient_theta(self):
+        if self._basis_inv_gradient_theta is None or self._basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return self._basis_inv_gradient_theta
+
+    @property
+    def basis_inv_gradient_phi(self):
+        if self._basis_inv_gradient_theta is None or self._basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return self._basis_inv_gradient_phi
+
+    def compute_inverse_gradient(self, inv_transform_type=None, weights=None):
+        if inv_transform_type != self.inv_transform_type:
+            self.inv_transform_type = inv_transform_type
+        if self.inv_transform_type is None:
+            ValueError("Inverse transform type not specified")
+        elif self.inv_transform_type == 'pseudo_inverse':
+            self._basis_inv_gradient_theta = np.linalg.pinv(self.basis_gradient_theta)
+            self._basis_inv_gradient_phi = np.linalg.pinv(self.basis_gradient_phi)
+        elif self.inv_transform_type == 'quadrature':
+            if weights is None:
+                weights = calculate_sampling_weights(self.coords)
+            self._basis_inv_gradient_theta = np.conj(self.basis_gradient_theta).T * (4 * np.pi * weights)
+            self._basis_inv_gradient_phi = np.conj(self.basis_gradient_phi).T * (4 * np.pi * weights)
+
+    def transform(self, signal):
+        if self.basis is None:
+            self.compute_basis()
+        return np.dot(self.basis.T, signal)
+
+    def transform_gradient(self, signal):
+        if self.basis_gradient_theta is None or self.basis_gradient_phi is None:
+            self.compute_basis_gradient()
+        return np.dot(self.basis_gradient_theta.T, signal), np.dot(self.basis_gradient_phi.T, signal)
+
+    def inverse_transform(self, coefficients):
+        if self.basis_inv is None:
+            self.compute_inverse()
+        return np.dot(self.basis_inv.T, coefficients)
+
+    def inverse_transform_gradient(self, coefficients):
+        if self.basis_inv_gradient_theta is None or self.basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return np.dot(self.basis_inv_gradient_theta, coefficients), np.dot(self.basis_inv_gradient_phi, coefficients)
+
+    def interpolate(self, non_uniform_coords):
+        # Implement interpolation algorithm using the basis matrix
+        # TODO: Implement interpolation algorithm using the basis matrix
+        pass
+
+    def filter(self, coefficients, band_type, cutoff_order):
+        # Implement filtering algorithm using the basis matrix
+        # TODO: Implement filtering
+        pass
+
+    def plot_basis_functions(self, order, mode):
+        # Implement plotting function for basis functions
+        # TODO: Implement plotting function for basis functions
+        pass
+
+    def plot_reconstructed_signal(self, coefficients):
```

Comment:
Does this require a dedictaed plotting function or could `pyfar.plot` be used for this?